### PR TITLE
SAK-48525 dashboard. Hide tasks widget by default.

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48525

### CODE REVIEW

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java [hunk 1]
```diff
@@ -103,38 +103,44 @@ public class DashboardController extends AbstractSakaiApiController {
...
     public void init() {
 
+        boolean tasksEnabled = serverConfigurationService.getBoolean("dashboard.tasks.enabled", false);
+
         // Load up all the available widgets, from properties
-        String[] courseWidgetsArray = serverConfigurationService.getStrings("dashboard.course.widgets");
-        if (courseWidgetsArray == null) {
-            courseWidgetsArray = new String[] {"tasks", "announcements", "calendar","forums", "grades" };
+        courseWidgets = serverConfigurationService.getStringList("dashboard.course.widgets", null);
+        if (courseWidgets.isEmpty()) {
```
Comment (ottenhoff) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:110]: will this survive a null?

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java [hunk 2]
```diff
@@ -103,38 +103,44 @@ public class DashboardController extends AbstractSakaiApiController {
...
+            courseWidgets = new ArrayList<>(List.of("tasks", "announcements", "calendar","forums", "grades"));
         }
-        courseWidgets = Arrays.asList(courseWidgetsArray);
+        if (!tasksEnabled) courseWidgets.remove("tasks");
 
-        String[] homeWidgetsArray = serverConfigurationService.getStrings("dashboard.home.widgets");
-        if (homeWidgetsArray == null) {
-            homeWidgetsArray = new String[] { "tasks", "announcements", "calendar","forums", "grades" };
+        homeWidgets = serverConfigurationService.getStringList("dashboard.home.widgets", null);
+        if (homeWidgets.isEmpty()) {
```
Comment (ottenhoff) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:116]: null?

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java (hunk 1 above)
Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:110]: Yep. getStringList returns an empty list if you supply a null default :)

Comment (ottenhoff) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:110]: haha ok thanks

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:110]: I've updated the javadoc on ServerConfigurationService to explain that

Comment (ottenhoff) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java:110]: that was nice of you. cheers.

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**